x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/create_datasource_page/index.tsx

```diff
@@ -95,32 +95,46 @@ export const CreateDatasourcePage: React.FunctionComponent = () => {
   // Datasource validation state
   const [validationResults, setValidationResults] = useState<DatasourceValidationResults>();
 
+  // Form state
+  const [formState, setFormState] = useState<DatasourceFormState>('INVALID');
+
   // Update package info method
-  const updatePackageInfo = (updatedPackageInfo: PackageInfo | undefined) => {
-    if (updatedPackageInfo) {
-      setPackageInfo(updatedPackageInfo);
-      setFormState('VALID');
-    } else {
-      setFormState('INVALID');
-      setPackageInfo(undefined);
-    }
+  const updatePackageInfo = useCallback(
+    (updatedPackageInfo: PackageInfo | undefined) => {
+      if (updatedPackageInfo) {
+        setPackageInfo(updatedPackageInfo);
+        if (agentConfig) {
+          setFormState('VALID');
+        }
+      } else {
+        setFormState('INVALID');
+        setPackageInfo(undefined);
+      }
 
-    // eslint-disable-next-line no-console
-    console.debug('Package info updated', updatedPackageInfo);
-  };
+      // eslint-disable-next-line no-console
+      console.debug('Package info updated', updatedPackageInfo);
+    },
+    [agentConfig, setPackageInfo, setFormState]
+  );
 
   // Update agent config method
-  const updateAgentConfig = (updatedAgentConfig: AgentConfig | undefined) => {
-    if (updatedAgentConfig) {
-      setAgentConfig(updatedAgentConfig);
-    } else {
-      setFormState('INVALID');
-      setAgentConfig(undefined);
-    }
+  const updateAgentConfig = useCallback(
+    (updatedAgentConfig: AgentConfig | undefined) => {
+      if (updatedAgentConfig) {
+        setAgentConfig(updatedAgentConfig);
+        if (packageInfo) {
+          setFormState('VALID');
+        }
+      } else {
+        setFormState('INVALID');
+        setAgentConfig(undefined);
+      }
 
-    // eslint-disable-next-line no-console
-    console.debug('Agent config updated', updatedAgentConfig);
-  };
+      // eslint-disable-next-line no-console
+      console.debug('Agent config updated', updatedAgentConfig);
```

Comment:
We can also use the logger from AppService now

Reply:
I believe `Logger` is only available as a server-side dependency

Reply:
👍 😿